drivers/ethernet/eth_adin2111.c

```diff
@@ -380,7 +391,7 @@ static int eth_adin2111_send_oa_frame(const struct device *dev, struct net_pkt *
 		}
 		if (i == chunks) {
 			hdr |= ADIN2111_OA_DATA_HDR_EV;
-			hdr |= (ctx->oa_cps - 1) << ADIN2111_OA_DATA_HDR_EBO;
+			hdr |= ebo << ADIN2111_OA_DATA_HDR_EBO;
```

Comment:
I think in this case (when we are at the last chunk), `cur` will need to be updated by adding `ebo` to it, instead of `cur += ctx->oa_cps;` on line 397.
Without this, the length of the frame passed is still having CPS (64) added to it, even if the EBO value is less.